src/registers.rs

```diff
@@ -203,8 +203,8 @@ pub(super) enum Register {
     T6 = 31,
 }
 
-impl From<u8> for Register {
-    fn from(value: u8) -> Register {
+impl Register {
+    pub(crate) const fn const_from(value: u8) -> Register {
```

Comment:
Looking forward to not having to do this when this lands: https://github.com/rust-lang/rust/issues/67792